README.md

```diff
@@ -113,6 +113,125 @@ function ApiRequestHandler(request, reply) {
 
 In the above example *isolated_db* is the same instance of TinyPg except with an overridden events property and *dispose* method to remove all listeners. The *UserService* can create other services and pass its reference to *isolated_db* to other services. In doing so, you can track all database queries executed as the result of every API request.
 
+## Hooks
+Hooks enable user defined functions to be called before (or after depending on the hook) TinyPg functions are executed. Most hooks allow user defined context to be passed throughout the hook lifecycle. Different hooks defined in the same `TinyHooks` object maintain a shared context across each hook call. The following hooks are currently supported:
```

Comment:
This documentation is really helpful! 

Given the ability to potentially modify the sql file or raw query, in `preSql` and `preRawQuery`, I think it would be helpful to explain the intent, restrictions, and possibilities with those hooks in particular since they are the only ones that can do anything other than modify the context.